src/com/interview/tree/IsBST.java

```diff
@@ -23,31 +26,65 @@
  */
 public class IsBST {
 
-    public boolean isBST(Node root){
-        return isBST(root, Integer.MIN_VALUE, Integer.MAX_VALUE);
-    }
-    
-    private boolean isBST(Node root, int min, int max){
-        if(root == null){
-            return true;
-        }
-        if(root.data <= min || root.data > max){
-            return false;
-        }
-        return isBST(root.left, min, root.data) && isBST(root.right, root.data, max);
-    }
-    
-    public static void main(String args[]){
-        BinaryTree bt = new BinaryTree();
-        Node root = null;
-        root = bt.addNode(10, root);
-        root = bt.addNode(15, root);
-        root = bt.addNode(-10, root);
-        root = bt.addNode(17, root);
-        root = bt.addNode(20, root);
-        root = bt.addNode(0, root);
-        
-        IsBST isBST = new IsBST();
-        assert isBST.isBST(root);
-    }
+	public boolean isBST(Node root) {
+		return this.isBST(root, Integer.MIN_VALUE, Integer.MAX_VALUE);
+	}
+
+	private boolean isBST(Node root, int min, int max) {
+		if (root == null) {
+			return true;
+		}
+		if ((root.data <= min) || (root.data > max)) {
+			return false;
+		}
+		return this.isBST(root.left, min, root.data)
+		        && this.isBST(root.right, root.data, max);
+	}
+
+	public boolean isBSTIterative(Node root) {
+		if (root == null) {
+			return true;
+		}
+		Deque<Node> stack = new LinkedList<Node>();
+		Node node = root;
+		int prev = node.data;
+		int current = 0;
+		while ( true ) {
+			if (node != null) {
+				stack.addFirst(node);
+				node = node.left;
+			} else {
+				if (stack.isEmpty()) {
+					break;
+				}
+				node = stack.pollFirst();
+				current = node.data;
+				prev = node.data;
```

Comment:
logic here looks wrong. Both current and prev will point to same value so prev > current will never happen